src/main/java/org/reaktivity/nukleus/maven/plugin/internal/generate/String16FlyweightGenerator.java

```diff
@@ -235,12 +246,24 @@ private MethodSpec setStringMethod()
                     .returns(stringType.nestedClass("Builder"))
                     .addParameter(String.class, "value")
                     .addParameter(Charset.class, "charset")
-                    .addStatement("byte[] charBytes = value.getBytes(charset)")
+                    .addStatement("byte[] charBytes")
+                    .addStatement("int length")
+                    .beginControlFlow("if (value == null)")
+                    .addStatement("charBytes = new byte[0]")
+                    .addStatement("length = -1")
+                    .nextControlFlow("else")
+                    .addStatement("charBytes = value.getBytes(charset)")
+                    .addStatement("length = charBytes.length")
+                    .endControlFlow()
+                    .beginControlFlow("if (length >= 65535)")
+                    .addStatement("throw new IllegalArgumentException("+
+                            "String.format(\"Value '%s' too big for String field (length < 65535)\", value))")
+                    .endControlFlow()
                     .addStatement("MutableDirectBuffer buffer = buffer()")
                     .addStatement("int offset = offset()")
                     .addStatement("int newLimit = offset + charBytes.length + FIELD_SIZE_LENGTH")
                     .addStatement("checkLimit(newLimit, maxLimit())")
-                    .addStatement("buffer.putShort(offset, (short) charBytes.length)")
+                    .addStatement("buffer.putShort(offset, (short) length)")
                     .addStatement("buffer.putBytes(offset + 2, charBytes)")
```

Comment:
Move `if` condition to simplify code, and also eliminate unneeded garbage via `new byte[0]`.

```
return methodBuilder("set")
.addModifiers(PUBLIC)
.returns(stringType.nestedClass("Builder"))
.addParameter(String.class, "value")
.addParameter(Charset.class, "charset")
.beginControlFlow("if (value == null)")
.addStatement("int newLimit = offset + FIELD_SIZE_LENGTH")
.addStatement("checkLimit(newLimit, maxLimit())")
.addStatement("buffer.putShort(offset, (short) -1)")
.addStatement("limit(newLimit)")
.nextControlFlow("else")
.addStatement("byte[] charBytes = value.getBytes(charset)")
.addStatement("int newLimit = offset + FIELD_SIZE_LENGTH + charBytes.length")
.addStatement("checkLimit(newLimit, maxLimit())")
.addStatement("buffer.putShort(offset, (short) charBytes.length)")
.addStatement("buffer.putBytes(offset + 2, charBytes)")
.addStatement("limit(newLimit)")
.endControlFlow()
.addStatement("valueSet = true")
.addStatement("return this")
.build();
```